tornado-api/src/main/java/uk/ac/manchester/tornado/api/collections/types/VectorDouble2.java

```diff
@@ -0,0 +1,232 @@
+
+import java.nio.DoubleBuffer;
+
+public class VectorDouble2 implements PrimitiveStorage<DoubleBuffer> {
+
+    /**
+     * backing array
+     */
+    final protected double[] storage;
+
+    /**
+     * number of elements in the storage
+     */
+    final private int numElements;
+    final private static int elementSize = 2;
+
+    /**
+     * Creates a vector using the provided backing array
+     *
+     * @param numElements
+     * @param array
+     */
+    protected VectorDouble2(int numElements, double[] array) {
+        this.numElements = numElements;
+        this.storage = array;
+    }
+
+    /**
+     * Creates a vector using the provided backing array
+     */
+    public VectorDouble2(double[] array) {
+        this(array.length / elementSize, array);
+    }
+
+    /**
+     * Creates an empty vector with
+     *
+     * @param numElements
+     */
+    public VectorDouble2(int numElements) {
+        this(numElements, new double[numElements * elementSize]);
+    }
+
+    private int toIndex(int index) {
+        return (index * elementSize);
+    }
+
+    /**
+     * Returns the float at the given index of this vector
+     *
+     * @param index
+     *
+     * @return value
+     */
+    public Double2 get(int index) {
+        return loadFromArray(storage, toIndex(index));
+    }
+
+    /**
+     * Sets the float at the given index of this vector
+     *
+     * @param index
+     * @param value
+     */
+    public void set(int index, Double2 value) {
+        value.storeToArray(storage, toIndex(index));
+    }
+
+    /**
+     * Sets the elements of this vector to that of the provided vector
+     *
+     * @param values
+     */
+    public void set(VectorDouble2 values) {
+        for (int i = 0; i < numElements; i++) {
+            set(i, values.get(i));
+        }
+    }
+
+    /**
+     * Sets the elements of this vector to that of the provided array
+     *
+     * @param values
+     */
+    public void set(double[] values) {
+        VectorDouble2 vector = new VectorDouble2(values);
+        for (int i = 0; i < numElements; i++) {
+            set(i, vector.get(i));
+        }
+    }
+
+    public void fill(float value) {
+        for (int i = 0; i < storage.length; i++) {
+            storage[i] = value;
+        }
+    }
+
+    /**
+     * Duplicates this vector
+     *
+     * @return
+     */
+    public VectorDouble2 duplicate() {
+        VectorDouble2 vector = new VectorDouble2(numElements);
+        vector.set(this);
+        return vector;
+    }
+
+    /**
+     * Prints the vector using the specified format string
+     *
+     * @param fmt
+     *
+     * @return
+     */
+    public String toString(String fmt) {
+        String str = "";
+
+        for (int i = 0; i < numElements; i++) {
+            str += get(i).toString() + " ";
+        }
+
+        return str;
+    }
+
+    /**
+     *
+     */
+    public String toString() {
+        if (numElements > 4) {
```

Comment:
why is this 4? shouldn't it be 2 (so, `elementSize`)?

Reply:
this is fixed now

Reply:
@mikepapadim this is not updated. is this on purpose?